src/provider/controller/ToastManager.ts

```diff
@@ -117,14 +117,15 @@ export class ToastManager extends AsyncInit {
     private async onAction(action: RootAction): Promise<void> {
         if (action instanceof CreateNotification) {
             this.create(action.notification);
-        }
-
-        if (action instanceof RemoveNotifications) {
+        } else if (action instanceof RemoveNotifications) {
             this.removeToasts(...action.notifications);
-        }
-
-        if (action instanceof ToggleCenterVisibility) {
+        } else if (action instanceof ToggleCenterVisibility) {
             this.closeAll();
+        } else if (action instanceof MinimizeToast) {
+            const toast = this._toasts.get(action.notification.id);
+            if (toast) {
```

Comment:
Since signals are async: If an API called `clear(note)` triggering a `RemoveNotifications` at the same time as the user clicks minimize firing this action.

Reply:
But there's still no async step between the dispatch and reaching this part of code. There's no promises that need to resolve before we get to here, and so no room for anything else to jump in, I think?

Reply:
There is no async step between dispatching and reaching this code, but other actions being handled by the `ToastManager` can modify `this._toasts`. When `MinimizeToast` is processed, there is no guarantee that the toast still exists.

Reply:
I'm gonna claim that if it exists in toast manager at the point the action is dispatched, then it exists in toast manager by the time we hit this line. How big an *if* that is, I'm not sure!